cocos/animation/exotic-animation/exotic-animation.ts

```diff
@@ -181,13 +181,12 @@ class ExoticNodeAnimation {
     private _scale: ExoticVec3Track | null = null;
 }
 
-function floatToHashString (value: number) {
+function floatToHashString (value: number): number {
     // Note: referenced to `Skeleton.prototype.hash`
-    return value.toPrecision(2);
+    return Number.parseFloat(value.toPrecision(2));
```

Comment:
Why does the function name `floatToHashString` return a `number` value instead of `string`?

Reply:
I've optimize this changed here
https://github.com/cocos/cocos-engine/pull/14572/commits/04a639dd12d6352e997ee1f2ae61e99ef082f326